src/main/java/org/broadinstitute/consent/http/service/ElasticSearchService.java

```diff
@@ -0,0 +1,180 @@
+    this.dataAccessRequestDAO = dataAccessRequestDAO;
+    this.dacDAO = dacDAO;
+    this.userDAO = userDAO;
+  }
+
+  public BulkResponse indexDatasets(List<DatasetTerm> datasets) throws IOException {
+    BulkRequest.Builder br = new BulkRequest.Builder();
+
+    for (DatasetTerm dataset : datasets) {
+      br.operations(op -> op
+          .index(idx -> idx
+              .index("datasets")
+              .id(dataset.getDatasetId().toString())
+              .document(dataset)
+          )
+      );
+    }
+
+    return esClient.bulk(br.build());
+  }
+
+  public UserTerm toUserTerm(User user) {
+
+    UserTerm term = new UserTerm();
+
+    term.setUserId(user.getUserId());
+    term.setDisplayName(user.getDisplayName());
+    if (Objects.nonNull(user.getRoles())) {
+      term.setRoles(user.getRoles().stream().map(UserRole::getName).toList());
+    }
+    term.setEmail(user.getEmail());
+
+    return term;
+  }
+
+  public DatasetTerm toDatasetTerm(Dataset dataset) {
+    DatasetTerm term = new DatasetTerm();
+
+    term.setDatasetId(dataset.getDataSetId());
+    term.setDatasetIdentifier(dataset.getDatasetIdentifier());
+
+    if (Objects.nonNull(dataset.getStudy())) {
+      Study study = dataset.getStudy();
+
+      term.setDescription(study.getDescription());
+      term.setStudyName(study.getName());
+      term.setStudyId(study.getStudyId());
+      term.setDataTypes(study.getDataTypes());
+      term.setPiName(study.getPiName());
+      term.setPublicVisibility(study.getPublicVisibility());
+
+      findStudyProperty(
+          study.getProperties(), "phenotypeIndication"
+      ).ifPresent(
+          prop -> term.setPhenotype((String) prop.getValue())
+      );
+
+      findStudyProperty(
+          study.getProperties(), "species"
+      ).ifPresent(
+          prop -> term.setSpecies((String) prop.getValue())
+      );
+
+      findStudyProperty(
+          study.getProperties(), "dataCustodianEmail"
+      ).ifPresent(
+          prop -> term.setDataCustodian((String) prop.getValue())
+      );
+
+      User dataSubmitter = userDAO.findUserById(study.getCreateUserId());
+      term.setDataSubmitter(toUserTerm(dataSubmitter));
+    }
+
+    if (Objects.nonNull(dataset.getDacId())) {
+      Dac dac = dacDAO.findById(dataset.getDacId());
+
+      if (Objects.isNull(dac)) {
+        throw new IllegalArgumentException("Could not find DAC");
+      }
+
+      term.setDacName(dac.getName());
+    }
+
+    Collection<Integer> approvedUserIds =
+        dataAccessRequestDAO.findAllUserIdsWithApprovedDARsByDatasetId(
+            dataset.getDataSetId());
+
+    if (!approvedUserIds.isEmpty()) {
+      Collection<User> approvedUsers = userDAO.findUsers(approvedUserIds);
+
+      List<UserTerm> approvedUserTerms = approvedUsers.stream().map(this::toUserTerm).toList();
+      term.setApprovedUsers(approvedUserTerms);
+    }
+
+    term.setDataUse(useRestrictionConverter.translateDataUseSummary(dataset.getDataUse()));
```

Comment:
per discussion w Greg, data use is saved as the output of `/translate/summary` on ontology, which might need more return fields. for example, with a DS restriction, there is no field which specifies the diseases it restricts to. Perhaps we want to add a ticket to do that?

Reply:
We do have a separate API to get the full disease term (see `/search` in Ontology) so we can still proceed with this PR and improve the efficiency with an update to the `/translate/summary` call.

Reply:
Apparently, there are 25 datasets in dev that have null data use objects so we need to handle that here too.